perf/newrawkey.c

```diff
@@ -70,25 +77,45 @@ int main(int argc, char *argv[])
         return EXIT_FAILURE;
     }
 
+    times = OPENSSL_malloc(sizeof(OSSL_TIME) * threadcount);
+    if (times == NULL) {
+        printf("Failed to create times array\n");
+        return EXIT_FAILURE;
+    }
+
     if (!perflib_run_multi_thread_test(do_newrawkey, threadcount, &duration)) {
         printf("Failed to run the test\n");
-        return EXIT_FAILURE;
+        goto out;
     }
 
     if (err) {
         printf("Error during test\n");
-        return EXIT_FAILURE;
+        goto out;
     }
 
-    us = ossl_time2us(duration);
-
-    avcalltime = (double)us / NUM_CALL_BLOCKS_PER_RUN;
+    us = times[0];
+    for (i = 1; i < threadcount; i++)
+        us = ossl_time_add(us, times[i]);
+    us = ossl_time_divide(us, NUM_CALLS_PER_TEST);
+
+    /*
+     * EVP_PKEY_new_raw_public_key is pretty fast, running in
+     * only a few us.  But ossl_time2us does integer division
+     * and so because the average us computed above is less than
+     * the value of OSSL_TIME_US, we wind up with truncation to
+     * zero in the math.  Instead, manually do the division, casting
+     * our values as doubles so that we comput the proper time
+     */
+    av = (double)ossl_time2ticks(us)/(double)OSSL_TIME_US;
 
     if (terse)
-        printf("%lf\n", avcalltime);
+        printf("%lf\n", av);
     else
-        printf("Average time per %d EVP_PKEY_new_raw_public_key_ex() calls: %lfus\n",
-               NUM_CALLS_PER_BLOCK, avcalltime);
+        printf("Average time per EVP_PKEY_new_raw_public_key_ex() calls: %lfus\n",
```

Comment:
Are these actually now `Average time per one EVP_PKEY_new_raw_public_key_ex() call:` and similarly below?

Reply:
Yes, thats my interpretation of how this works now:
1) Each spawned thread marks the start  time of their loop
2) Each thread calls EVP_PKEY_new_raw_public_key_ex NUM_CALLS_PER_TEST/threadcount times
3) Each spawned thread marks the end time of their loop, subtracts the start time from it, and records it
4) When all threads are joined, the main routine adds each threads duration (from (2)), and divides that aggregate  value by NUM_CALLS_PER_TEST

As such SUM(i=0..threadcount, thread_duration[i]) / NUM_CALLS_PER_TEST should be the average call time for a single call to EVP_PKEY_new_raw_public_key

Reply:
So, could you please change the text message as the plural there does not make sense to me anymore.

Reply:
handled, i audited the tests and removed the plural from tests, as they all measure the average single call time now

Reply:
You didn't seem to push the commits?